Gemfile

```diff
@@ -5,3 +5,5 @@ gemspec
 
 gem "rake", "~> 12.0"
 gem "minitest", "~> 5.0"
+
+gem "byebug", "~> 11.1", :group => :development
```

Comment:
Gems that are used inside another gem like we have here are actually done a little differently from how they are within a Rails app. Within a gem, using "group" like we do here doesn't have the same effect because unlike a Rails app, we have on concept of "environment". Our library might be used within a Rails app, or might not!

What we'll want to do instead is add it as a development dependency within the `gemspec` file. A gemspec is the way a gem declares information about itself, including it's dependencies. If you notice, it's also called at the top of this Gemfile to include those dependencies when bundling.

[Here is an example of adding a development dependency to a gem](https://bundler.io/guides/creating_gem.html#using-other-gems).

In our case, we'll want to open up the degreed.gemspec file and add `spec.add_development_dependency "byebug"` and remove the entry for byebug here.

A good question would also be, why are "rake" and "minitest" in here instead of the gemspec? That is a good question... I'm not sure actually haha!! It works because any gems in here override the gemspec, but the appropriate place to define a gems dependencies is in the gemspec; so I'm not sure why there are here (I think the generator for the gem might have added them? But why?)

Reply:
The more you know! Thanks @rreinhardt9. 

I am now wondering why "rake" and "minitest" are in the Gemfile when like you mentioned. 

Is it worth adding these to the gemspec file to follow that convention? 🤔

Reply:
> Is it worth adding these to the gemspec file to follow that convention?

I'm fine leaving them as is in this PR... might be something to think about in the future though. I'd like to try and understand why they started there before I moved them 🙃